### PR TITLE
Created fibonacci.cpp

### PR DESCRIPTION
**Pull request by miishh**
Created a function called "fib" to calculate the nth fibonacci number in c++ using dp reducing the time complexity of the problem to O(n).

**Fixes issue:**

Solution of fibonacci in c++.

**Changes:**

Wrote whole function in c++ to find nth fibonacci no using dp.

### CODE REVIEW

File: code/dynamic_programming/src/fibonacci/fibonacci.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#include <iostream>
+using namespace std;
+
+int fib(int n)
+{
+    int *ans = new int[n + 1]; //creating an array to store the outputs
+    ans[0] = 0;
+    ans[1] = 1;
+    for (int i = 23; i <= n; i++)
```
Comment (vishalg8454) [inline code/dynamic_programming/src/fibonacci/fibonacci.cpp:9]: typo in loop, should be `for(int i=2; i<=n; i++)`

File: code/dynamic_programming/src/fibonacci/fibonacci.cpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        ans[i] = ans[i - 1] + ans[i - 2];  // storing outputs for further use
+    }
+    return ans[n];
+}
+int main()
+{
+    int n;
+    cout << " Enter the Number";  // taking input
```
Comment (vishalg8454) [inline code/dynamic_programming/src/fibonacci/fibonacci.cpp:18]: Remove the leading space and add semicolon at the end for better output display
`cout << "Enter the Number: ";`

**Review by vishalg8454 — CHANGES_REQUESTED** [review]

File: code/dynamic_programming/src/fibonacci/fibonacci.cpp (hunk 2 above)
Comment (miishh) [reply code/dynamic_programming/src/fibonacci/fibonacci.cpp]: ok sir

**Review by AdiChat — DISMISSED** [review]

**Review by AdiChat — APPROVED** [review]

**Comment by AdiChat** [comment]
Looks great 😍 

You have made a wonderful contribution. 
Join us [**at our community forum**](https://discourse.opengenus.org/signup) :family_man_woman_girl_boy: to get connected and access to community benefits

If you would like to [join our internship program](https://discourse.opengenus.org/t/internship-guidelines-at-opengenus/2335/) and contribute to [OpenGenus IQ](https://iq.opengenus.org/), let us know and we will be happy to have you onboard 

Hope you are enjoying your journey 🚆  with **OpenGenus**
positive_opengenus

**Pull request merged**